File: head first OOAD Guitar/head first OOAD Guitar/Inventory.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace head_first_OOAD_Guitar
{
    class Inventory
    {
        private List<object> guitars;
        public Inventory()
        {
            guitars = new List<object>();
        }
        public void addGuitar(String serialNumber, double price, GuitarSpec spec)
        {
            Guitar guitar = new Guitar(serialNumber, price, spec);
            guitars.Add(guitar);
        }
        public Guitar getGuitar(String serialNumber)
        {
            int index = 0;
            foreach(var i in guitars)
            {
                Guitar guitar = (Guitar)guitars[index];
                if (guitar.getSerialNumber().Equals(serialNumber))
                {
                    return guitar;
                }
                index++;
            }
            return null;
        }

        public List<object> search(GuitarSpec searchSpec)
        {
            List<object> matchingGuitars = new List<object>();
            int index = 0;
            foreach(var i in guitars)
            {
                Guitar guitar = (Guitar)guitars[index];
                if(guitar.GetSpec().matches(searchSpec))
                    matchingGuitars.Add(guitar);
                index++;
            }
            return matchingGuitars;
        }
    }
}

File: head first OOAD Guitar Python/head first OOAD Guitar/Inventory.py
from Guitar import Guitar
from GuitarSpec import GuitarSpec

staticList1 = []#use static list
staticList2 = []#declare different variable to assign avoid append matchingGuitar, also append self._guitars
class Inventory(object):
    def __init__(self):
        self._guitars = staticList1

    def addGuitar(self,serialNumber,price,spec):
        self.guitar = Guitar(serialNumber,price,spec)
        self._guitars.append(self.guitar)

    def getGuitar(self,serialNumber):
        index = 0
        for i in self._guitars:
            self.guitar = self._guitars[index]
            if self.guitar.getSerialNumber() == serialNumber:
                return self.guitar
            index += 1
        return None

    def search(self,searchSpec):
        self.matchingGuitars = staticList2
        index = 0
        for i in self._guitars:
            self.guitar = self._guitars[index]
            if self.guitar.GetSpec().matches(searchSpec):
                self.matchingGuitars.append(self.guitar)
            index+=1
        return self.matchingGuitars